lookout/style/format/features.py

```diff
@@ -99,17 +98,40 @@ class FeatureExtractor:
     def __init__(self, language: str):
         self.tokens = importlib.import_module("lookout.style.format.langs.%s.tokens" % language)
         self.roles = importlib.import_module("lookout.style.format.langs.%s.roles" % language)
+        self.feature_names = ['start_offset', 'start_line', 'start_col', 'end_offset', 'end_line',
+                              'end_col', 'internal_role']
 
-    def extract_features(self, files: List[File]):
+    def extract_features(self, files: List[File], parents_depth: int = 2,
+                         siblings_window: int = 5) -> Tuple[numpy.ndarray, numpy.ndarray]:
         """
         This is the dream interface for the feature extraction.
 
         :param files: the list of `File`-s (see service_data.proto) of the same language.
-        :param language: the name of the language.
-        :param config: feature extraction parameters.
-        :return:
+        :param parents_depth: how many parents to use for each node.
+        :param siblings_window: how many siblings to use for each node (both left and right).
+        :return: tuple of numpy.ndarray (2 and 1 dimensional respectively): features and labels.
         """
-        pass
+        parsed_files = []
+        labels = []
+        for file in files:
+            contents = file.content.decode('utf-8')
```

Comment:
We use double quotes https://github.com/src-d/guide/blob/master/engineering/conventions-python.md#code-style

Reply:
From my experience we will definitely hit UnicodeDecodeError-s here.

```python
content.decode("utf-8", "replace")
```

The replacement is needed to not interfere with the offsets.